tests/functional/test_s3.py

```diff
@@ -497,6 +513,35 @@ def test_correct_url_used_for_s3():
             'https://bucket.s3-accelerate.amazonaws.com/key'))
     yield t.case(
         region='us-east-1', bucket='bucket', key='key',
+        # Dualstack is whitelisted.
+        customer_provided_endpoint=(
+            'https://s3-accelerate.dualstack.amazonaws.com'),
+        expected_url=(
+            'https://bucket.s3-accelerate.dualstack.amazonaws.com/key'))
+    yield t.case(
+        region='us-east-1', bucket='bucket', key='key',
+        # Even whitelisted parts cannot be duplicated.
+        customer_provided_endpoint=(
+            'https://s3-accelerate.dualstack.dualstack.amazonaws.com'),
+        expected_url=(
+            'https://s3-accelerate.dualstack.dualstack'
+            '.amazonaws.com/bucket/key'))
+    yield t.case(
+        region='us-east-1', bucket='bucket', key='key',
+        # More than two extra parts is not allowed.
```

Comment:
If this for more than two extra parts maybe not use ``dualstack`` as you already have a test for double ``dualstack``.